suite-common/wallet-core/src/fiat-rates/fiatRatesThunks.ts

```diff
@@ -297,6 +300,61 @@ export const updateLastWeekFiatRatesThunk = createThunk(
     },
 );
 
+/**
+ * Updates the price data for the past 14 days in 1-day interval (14 data points)
+ */
+export const updateLastTwoWeekFiatRatesThunk = createThunk(
```

Comment:
Yes good point, I can make this more generic with weeks argument.

Btw, are you talking about the existing `updateLastWeekFiatRatesThunk` too? Because this does something a little bit different and I didn't want to break this feature anywhere..
Thing is, `updateLastWeekFiatRatesThunk` gets the data for 1 day only (7 days ago) for all fiat currencies, so the description of this function is wrong (it doesn't get 168 data points, only 1).
My first thought initially was to have _updateLast**Two**WeekFiatRatesThunk_ only and calculate `lastWeek` object from this data (to eliminate duplicate API calls), but `lastWeek` has rates for all fiat currencies and for this new TradeBox component we need rates for 14 days in past, which might be a lot of data for all fiat currencies (10+ KB), so instead of fetching the data for all currencies I made this `localCurrency` argument which fetch the data only for a single fiat currency which is a lot more smaller in size.
I'm not sure about this really, what do you think?

Reply:
I am and yeah, I've noticed that it gets 1 data point and wasn't sure if it's just bad naming or a bug. Well, is there a reason to get data for all fiat currencies though? 🤔 It seems like `localCurrency` is always used to access the rates, and those rates can be refetched whenever it changes. And then, as you say, everything can be fetched once to be used in both places :)

Reply:
Yea it should be refetched in case of localCurrency changed, I wasn't sure really why it was designed this way and if I'm not missing something..
I'll try to merge them into one then

Reply:
https://github.com/trezor/trezor-suite/pull/9722/commits/1f640b276ac910729494a8aeae9441d96ecf2dbd Here I combined them into a single API request and eliminated some of the duplicities, in state there is still lastWeek and lastTwoWeeks though, what do you think?